### ISSUE TITLE
Cannot go back to the editor after running project in the Android editor

### ISSUE THREAD

**Issue by iqbal70** [godotengine/godot#79291]
### Godot version

4.1.0.stable

### System information

Godot v4.1.stable - Android - Vulkan (Compatibility) - PowerVR Rogue GE8300 -  (4 Threads)

### Issue description

Expected: When The Back Button Is Pressed The Game Should Stop/Killed And Back To The Editor

What Happen: When The Back Button Is Pressed You Back To The Home Screen And If You Open The App From Recent App You Will Notice That The Name Of The App Is Changed And The Game Is Runned Again And Its Impossible To Back To The Editor Unless You Close The App And ReOpen The Editor 

https://github.com/godotengine/godot/assets/73569682/3b8a197e-5be1-4ee4-b546-44baad280c84

Again

### Steps to reproduce

1. Create A New Project
2. Run The Project
3. Press Back Button Or Quit Button

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot/issues/61611, which was supposed to be fixed by https://github.com/godotengine/godot/pull/72106.

**Comment by jojorne** [comment]
If it was fixed I can confirm that this issue is back. Hitting back makes the whole editor quit instead of returning to the editor.

**Comment by EarthDLL** [comment]
In fact, I tested on three devices with version 4.2.beta1

Firstly, Android 7 only supports opengl3 and cannot open any projects, resulting in editor crashes. Then I tried version 4.1.2 and clicked the return button to return to the editor correctly. But when running the project, if you switch to the background and then return to the front end, and click the return button, this bug will appear again.

Secondly, for Android 8, the situation is the same as the previous one. Once the project is switched to the backend while running, bugs will appear.

Thirdly, Android 10, this bug will occur in any situation and there is no solution.(both 4.1.2 and 4.2beta1)
